Gems/EMotionFX/Code/EMotionFX/CommandSystem/Source/JointLimitCommands.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/std/smart_ptr/make_shared.h>
+#include <EMotionFX/CommandSystem/Source/CommandManager.h>
+#include <EMotionFX/CommandSystem/Source/JointLimitCommands.h>
+#include <EMotionFX/Source/Actor.h>
+#include <EMotionFX/Source/ActorManager.h>
+#include <EMotionFX/Source/Allocators.h>
+#include <EMotionFX/Source/EMotionFXManager.h>
+#include <EMotionFX/Source/Node.h>
+#include <EMotionFX/Source/PhysicsSetup.h>
+#include <MCore/Source/ReflectionSerializer.h>
+
+namespace EMotionFX
+{
+    AZ_CLASS_ALLOCATOR_IMPL(ParameterMixinActorIdJointName, EMotionFX::CommandAllocator, 0);
+    AZ_CLASS_ALLOCATOR_IMPL(CommandAdjustJointLimit, EMotionFX::CommandAllocator, 0);
+
+    const char* CommandAdjustJointLimit::CommandName = "AdjustJointLimit";
+
+    ParameterMixinActorIdJointName::ParameterMixinActorIdJointName(AZ::u32 actorId, const AZStd::string& jointName)
+        : ParameterMixinActorId(actorId)
+        , ParameterMixinJointName(jointName)
+    {
+    }
+
+    void ParameterMixinActorIdJointName::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (!serializeContext)
+        {
+            return;
+        }
+
+        serializeContext->Class<ParameterMixinActorIdJointName, ParameterMixinActorId, ParameterMixinJointName>()->Version(1);
+    }
+
+    AzPhysics::JointConfiguration* CommandAdjustJointLimit::GetJointConfiguration(Actor** outActor, AZStd::string& outResult) const
+    {
+        Actor* actor = GetActor(this, outResult);
+        if (!actor)
+        {
+            *outActor = nullptr;
+            return nullptr;
+        }
+        *outActor = actor;
+
+        const AZStd::shared_ptr<PhysicsSetup>& physicsSetup = actor->GetPhysicsSetup();
+        if (physicsSetup)
+        {
+            const Physics::RagdollConfiguration& ragdollConfig = physicsSetup->GetRagdollConfig();
+            const Physics::RagdollNodeConfiguration* ragdollNodeConfig = ragdollConfig.FindNodeConfigByName(m_jointName);
+            return ragdollNodeConfig ? ragdollNodeConfig->m_jointConfig.get() : nullptr;
+        }
+        return nullptr;
+    }
+
+    CommandAdjustJointLimit::CommandAdjustJointLimit(MCore::Command* orgCommand)
+        : MCore::Command(CommandName, orgCommand)
+        , m_oldIsDirty(false)
```

Comment:
We should be able to remove this as we're default-initing that in the header already.

Reply:
OK, done.